src/ValidationError.js

```diff
@@ -11,5 +13,41 @@ const ValidationError = function(message, type, data) {
 
 ValidationError.prototype = Object.create(Error.prototype);
 ValidationError.constructor = ValidationError;
+ValidationError.JoiFactory = function(errorDetails) {
```

Comment:
You're doing way more here than what is described in the issue. I think that using `Joi.extend` is probably a good idea (instead of relying on custom handling in our apis which is what we do now) I don't think I'd implement it quite this way.

Hacking JoiFactory into the the ValidationError object isn't my favorite thing and certainly is out of the scope of this issue.